fleetshard/pkg/central/charts/data/rhacs-operator/templates/rhacs-operator-deployment.yaml

```
@@ -1,10 +1,14 @@
+## Iterate over operator versions passed by fleet-shard sync
+{{- range .Values.operator.images }}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
   labels:
     app: rhacs-operator
     control-plane: controller-manager
-  name: rhacs-operator-controller-manager
+  ## Name field must contain up to 63 characters
+  ## https://www.rfc-editor.org/rfc/rfc1123
+  name: rhacs-operator-controller-manager-{{ .tag | trunc 10 }}-{{ randAlphaNum 10 | lower }}
```

minor:
```suggestion
  name: rhacs-operator-controller-manager-{{ .tag | trunc 10 | lower }}-{{ randAlphaNum 10 | lower }}
```

---

Just a quick thought - doesn't using random strings cause all deployments to be replaced on every run? I don't think we want that. Why not just fail on longer tags? I don't think we'll ever need them :thinking:

---

@porridge I agree. Also since the "tag" is our "unique" key to install operator versions, we don't need a random suffix

---

I've changed it in a way that long tags will be ignored. My only concern is that we probably should not return an error if there is one too long tag version but instead keep parsing the next image. I am not 100% sure about it though.

Let's assume that we have received two operator images for deployment. One is fine and another is too long to be the suffix for deployment. Should we cancel both deployment or only one with long version?

---

I don't think we should be silently ignoring requests to install a problematic operator. These cases always make me think of a compiler which just ignores the statements that it didn't understand, because it wants to be robust and produce a program that works "mostly" :-)

---

I added fail fast on parsing images